---
title: "Lab10"
author: "Anastasia Galperina"
date: "11/2/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Part I - Reading VCF data
```{r}
library(vcfR)
library(adegenet)
library(pinfsc50)
library(reshape2)
library(ggplot2)
```

```{r}
data(vcfR_example)
vcf
```

```{r}
strwrap(vcf@meta[1:7])
```

```{r}
queryMETA(vcf)
```

```{r}
queryMETA(vcf, element = 'DP')
```

```{r}
queryMETA(vcf, element = 'FORMAT=<ID=DP')
```

```{r}
head(getFIX(vcf))
```

```{r}
vcf@gt[1:6, 1:4]
```

```{r}
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz")
head(vcf)
```

```{r, eval = FALSE}
write.vcf(vcf, "myVCFdata_filtered.vcf.gz")
```

## Exercises Part I
# Find more information about read.vcfR()
```{r}
?read.vcfR
```

# Learn what the acronym “AD” stands for
```{r}
queryMETA(vcf, element = 'AD')
```

# View the last few lines of fix data
```{r}
tail(vcf@fix)
```

# Does QUAL appear useful to us?
```{r}
plot(vcf)
```

```{r, eval = FALSE}
# Alternate ggplot2 answer.
library(ggplot2)
qplot(getQUAL(vcf), geom = "histogram")
```

# Query the sample names
```{r}
colnames(vcf@gt)
```

## Part II - Analysis of Genome Data
```{r}
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz")
head(vcf)
```

```{r}
x <- vcfR2genlight(vcf)
x
```

```{r}
gt <- extract.gt(vcf, element = "GT")
gt[c(2,6,18), 1:3]
```

```{r}
t(as.matrix(x))[c(1,5,17), 1:3]
```

```{r}
pop(x) <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
popNames(x)
```

```{r}
ploidy(x) <- 2
```

```{r}
x.dist <- dist(x)
```

```{r}
x.dist <- poppr::bitwise.dist(x)
```

```{r}
# Find the files.
vcf_file <- system.file("extdata", "pinf_sc50.vcf.gz", package = "pinfsc50")
dna_file <- system.file("extdata", "pinf_sc50.fasta", package = "pinfsc50")
gff_file <- system.file("extdata", "pinf_sc50.gff", package = "pinfsc50")

# Input the files.
vcf <- read.vcfR(vcf_file, verbose = FALSE)
dna <- ape::read.dna(dna_file, format = "fasta")
gff <- read.table(gff_file, sep="\t", quote="")

# Create a chromR object.
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=TRUE)
chrom

plot(chrom)
# read depth here is a sum over all samples
# SNP densities is empty because this data is created during the processing of chromR objects
```

```{r}
chromoqc(chrom, dp.alpha = 66)
# displays the same information as the plot method only it distributes the data along its chomosomal coordinates
```

```{r}
chrom <- proc.chromR(chrom, verbose = TRUE)
plot(chrom)
```

```{r}
chromoqc(chrom, dp.alpha = 66)
```

```{r}
vcf <- read.vcfR("data/pinfsc50_filtered.vcf.gz", verbose = FALSE)
chrom <- create.chromR(name="Supercontig", vcf=vcf, seq=dna, ann=gff, verbose=FALSE)
chrom <- proc.chromR(chrom, verbose = FALSE)
chromoqc(chrom, dp.alpha = 66)
```

```{r}
head(chrom@var.info)
```

```{r}
head(chrom@win.info)
```

```{r}
data(vcfR_example)
pop <- as.factor(c("us", "eu", "us", "af", "eu", "us", "mx", "eu", "eu", "sa", "mx", "sa", "us", "sa", "Pmir", "us", "eu", "eu"))
myDiff <- genetic_diff(vcf, pops = pop, method = 'nei')
knitr::kable(head(myDiff[,1:15]))
```

```{r}
knitr::kable(head(myDiff[,16:19]))
```

```{r}
knitr::kable(round(colMeans(myDiff[,c(3:9,16,19)], na.rm = TRUE), digits = 3))
```

```{r}
dpf <- melt(myDiff[,c(3:8,19)], varnames=c('Index', 'Sample'), value.name = 'Depth', na.rm=TRUE)
```

```{r}
p <- ggplot(dpf, aes(x=variable, y=Depth)) + geom_violin(fill="#2ca25f", adjust = 1.2)
p <- p + xlab("")
p <- p + ylab("")
p <- p + theme_bw()
p
```

# Exercises Part II
# Manhattan plot
```{r}
plot(getPOS(vcf), myDiff$Gprimest,  pch = 20, col = "#1E90FF44", xlab = "", ylab = "", ylim = c(0, 1), xaxt = "n")
axis(side = 1, at = seq(0, 1e5, by = 1e4), labels = seq(0, 100, by = 10))
title(xlab='Genomic position (Kbp)')
title(ylab = expression(italic("G'"["ST"])))
```

# What is wrong with plot
```{r}
# small sample size
# check with 
table(pop)
```

# Zoom in on a particular region of a chromosome
```{r}
chromoqc(chrom, dp.alpha = 66, xlim = c(2e05, 4e05))
```

# Look for other data in your file that may be of interest
```{r}
queryMETA(vcf)
```

# References 
References
Danecek, Petr, Adam Auton, Goncalo Abecasis, Cornelis A Albers, Eric Banks, Mark A DePristo, Robert E Handsaker, et al. 2011. “The Variant Call Format and VCFtools.” Bioinformatics 27 (15): 2156–8. https://doi.org/10.1093/bioinformatics/btr330.

Grünwald, Niklaus J, Bruce A McDonald, and Michael G Milgroom. 2016. “Population Genomics of Fungal and Oomycete Pathogens.” Annual Review of Phytopathology 54: 323–46. https://doi.org/0.1146/annurev-phyto-080614-115913.

Hedrick, Philip W. 2005. “A Standardized Genetic Differentiation Measure.” Evolution 59 (8): 1633–8. http://dx.doi.org/10.1111/j.0014-3820.2005.tb01814.x.

Jombart, Thibaut. 2008. “adegenet
: A R Package for the Multivariate Analysis of Genetic Markers.” Bioinformatics 24 (11): 1403–5. https://doi.org/10.1093/bioinformatics/btn129.

Jost, Lou. 2008. “GST
 And Its Relatives Do Not Measure Differentiation.” Molecular Ecology 17 (18): 4015–26. http://dx.doi.org/10.1111/j.1365-294X.2008.03887.x.

Kamvar, Zhian N, Jonah C Brooks, and Niklaus J Grünwald. 2015. “Novel R tools for analysis of genome-wide population genetic data with emphasis on clonality.” Name: Frontiers in Genetics 6: 208. https://doi.org/10.3389/fgene.2015.00208.

Kamvar, Z N, J F Tabima, and Niklaus J Grünwald. 2014. “Poppr
: An R Package for Genetic Analysis of Populations with Clonal, Partially Clonal, and/or Sexual Reproduction.” PeerJ 2: e281. https://doi.org/10.7717/peerj.281.

Knaus, Brian J, and Niklaus J Grünwald. 2017. “Vcfr
: A Package to Manipulate and Visualize Variant Call Format Data in R.” Molecular Ecology Resources 17 (1): 44–53. http://dx.doi.org/10.1111/1755-0998.12549.

Luikart, Gordon, Phillip R England, David Tallmon, Steve Jordan, and Pierre Taberlet. 2003. “The Power and Promise of Population Genomics: From Genotyping to Genome Typing.” Nature Reviews Genetics 4 (12): 981–94. https://doi.org/10.1038/nrg1226.

Nei, Masatoshi. 1973. “Analysis of Gene Diversity in Subdivided Populations.” Proceedings of the National Academy of Sciences 70 (12): 3321–3. http://www.pnas.org/content/70/12/3321.abstract.

Paradis, Emmanuel, Julien Claude, and Korbinian Strimmer. 2004. “APE: Analyses of Phylogenetics and Evolution in R Language.” Bioinformatics 20 (2): 289–90. https://doi.org/10.1093/bioinformatics/btg412.

Paradis, Emmanuel, Thierry Gosselin, Niklaus J Grünwald, Thibaut Jombart, Stéphanie Manel, and Hilmar Lapp. 2017. “Towards an Integrated Ecosystem of R Packages for the Analysis of Population Genetic Data.” Molecular Ecology Resources 17 (1): 1–4. https://doi.org/10.1111/1755-0998.12636.

Simpson, Edward H. 1949. “Measurement of Diversity.” Nature 163: 688. http://dx.doi.org/10.1038/163688a0.

Wright, Sewall. 1949. “The Genetical Structure of Populations.” Annals of Eugenics 15 (1): 323–54. https://doi.org/10.1111/j.1469-1809.1949.tb02451.x.